Sources/Afluent/Workers/SingleValueSubject.swift

```diff
@@ -30,17 +30,26 @@ public final class SingleValueSubject<Success: Sendable>: AsynchronousUnitOfWork
         AsynchronousOperation { [weak self] in
             guard let self else { throw CancellationError() }
 
-            self.lock()
-            if case .sentValue(let success) = self.subjectState {
-                self.unlock()
-                return success
-            } else if case .sentError(let error) = self.subjectState {
-                self.unlock()
-                throw error
+            func getSentValue() throws -> Success? {
+                defer { self.unlock() }
```

Comment:
Can you use `lock.protect { }` here? It's available in an extension

Reply:
Sure thing